src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -331,6 +334,20 @@ internal virtual LexicalSortKey GetLexicalSortKey()
         /// </summary>
         public abstract ImmutableArray<Location> Locations { get; }
 
+#nullable enable
+
+        public virtual Location? TryGetFirstLocation()
```

Comment:
nit. `Try` tends to imply a bool return + out parameter. Is that fine?

Reply:
we use that name to indicate 0/1 return as well :)

Reply:
> Try tends to imply a bool return + out parameter. Is that fine?

I would say that this pattern is used when return value cannot reliably convey the fact of failure. In this case it can.